src/driver/mpas_subdriver.F

```diff
@@ -426,6 +440,12 @@ subroutine mpas_finalize(corelist, domain_ptr)
       !
       call MPAS_stream_mgr_finalize(domain_ptr % streamManager)
 
+      if (domain_ptr % streamInfo % finalize() /= 0) then
```

Comment:
The `nvhpc` build points to this line as a source of the Internal Compiler error.

Part of my build log:
```
mpifort -D_MPI -DCPRPGI -DCORE_ATMOSPHERE -DMPAS_NAMELIST_SUFFIX=atmosphere -DMPAS_EXE_NAME=atmosphere_model - DMPAS_DEBUG -DSINGLE_PRECISION -DMPAS_NATIVE_TIMERS -DMPAS_GIT_VERSION=v8.0.1-58-gf642c4ea6-dirty -            DMPAS_BUILD_TARGET=nvhpc -DMPAS_SMIOL_SUPPORT -O0 -g -Mbounds -Mchkptr -byteswapio -Mfree -Ktrap=divz,fp,inv,  ovf -traceback -c mpas_subdriver.F -I/glade/u/apps/derecho/23.06/spack/opt/spack/netcdf/4.9.2/cray-mpich/8.1.  25/nvhpc/23.1/o74c/include -I/glade/u/apps/derecho/23.06/spack/opt/spack/parallel-netcdf/1.12.3/cray-mpich/8.  1.25/nvhpc/23.1/qzqm/include -I/glade/work/gdicker/mpas-work/PullRequestTesting/2023Dec31_MPASAPR1132/MPAS-    Model/src/external/SMIOL -I/glade/u/apps/derecho/23.06/spack/opt/spack/netcdf/4.9.2/cray-mpich/8.1.25/nvhpc/   23.1/o74c/include -I/glade/u/apps/derecho/23.06/spack/opt/spack/parallel-netcdf/1.12.3/cray-mpich/8.1.25/      nvhpc/23.1/qzqm/include -I../framework -I../operators -I../core_atmosphere -I../external/esmf_time_f90
nvfortran-Warning-CUDA Fortran or OpenACC GPU targets disables -Mbounds
NVFORTRAN-S-0000-Internal compiler error. check_member: member arrived with wrong derived type    5621         (mpas_subdriver.F: 443)
 0inform,   0 warnings,   1 severes, 0 fatal for mpas_finalize
make[4]: *** [Makefile:24: mpas_subdriver.o] Error 2
```

Reply:
I've just added a commit to this PR (f41bedae) to work around what at first glance appears to be a compiler bug. Just as a sanity check, I also verified that the code compiled (before the additional commit) with the Cray compilers, in addition to the Intel and GNU compilers.